app/src/main/java/com/kks/nimblesurveyjetpackcompose/repo/login/LoginRepoImpl.kt

```diff
@@ -0,0 +1,63 @@
+package com.kks.nimblesurveyjetpackcompose.repo.login
+
+import com.kks.nimblesurveyjetpackcompose.di.ServiceQualifier
+import com.kks.nimblesurveyjetpackcompose.model.ApiInterface
+import com.kks.nimblesurveyjetpackcompose.model.ResourceState
+import com.kks.nimblesurveyjetpackcompose.model.request.LoginRequest
+import com.kks.nimblesurveyjetpackcompose.model.response.LoginResponse
+import com.kks.nimblesurveyjetpackcompose.util.*
+import com.kks.nimblesurveyjetpackcompose.util.extensions.safeApiCall
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.flow
+import javax.inject.Inject
+
+class LoginRepoImpl @Inject constructor(
+    @ServiceQualifier private val apiInterface: ApiInterface,
+    private val preferenceManager: PreferenceManager,
+    private val customKeyProvider: CustomKeyProvider
+) : LoginRepo {
+
+    override fun loginWithEmailAndPassword(
+        email: String,
+        password: String
+    ): Flow<ResourceState<LoginResponse>> =
+        flow {
```

Comment:
If you one-line this we could reduce indention 😉 
```suggestion
    ): Flow<ResourceState<LoginResponse>> = flow {
```